File: src/types/analytics.ts
// ── Analytics types ──────────────────────────────────────────────────────────
import type { TransactionCategory } from './transaction';

export interface AnalyticsSummary {
  currentMonthIncome: number;
  previousMonthIncome: number;
  incomeChange: number;           // percentage
  currentMonthExpenses: number;
  previousMonthExpenses: number;
  expensesChange: number;         // percentage
}

export interface CategoryBreakdown {
  category: TransactionCategory;
  amount: number;
  percentage: number;
  previousAmount?: number;
}

export interface MonthlyTrendPoint {
  month: string;   // e.g. "Jan", "Feb"
  income: number;
  expenses: number;
  savings: number;
}

export interface AIInsight {
  id: string;
  type: 'spending_insight' | 'saving_insight' | 'warning' | 'recommendation';
  title: string;
  message: string;
  generatedAt: string;
}

File: backend/app/schemas/analytics.py
from pydantic import BaseModel
from typing import List, Optional


class AnalyticsSummaryOut(BaseModel):
    currentMonthIncome: float
    previousMonthIncome: float
    incomeChange: float
    currentMonthExpenses: float
    previousMonthExpenses: float
    expensesChange: float


class CategoryBreakdownOut(BaseModel):
    category: str
    amount: float
    percentage: float
    previousAmount: Optional[float] = None


class MonthlyTrendPointOut(BaseModel):
    month: str
    income: float
    expenses: float
    savings: float


class AIInsightOut(BaseModel):
    id: str
    type: str
    title: str
    message: str
    generatedAt: str
